Documentation/DomainModel/Entities.md

```diff
@@ -0,0 +1,209 @@
+
+This contains records of changes to a **subscriber** account. In phpList 3,
+the trail is accessible on the subscriber history page, e.g.,
+/admin/?page=userhistory&id=1. It includes when and how they were subscribed
+and reasons for their blacklisting.
+
+### Subscription
+Table name: `phplist_listuser`
+
+A **subscription** means that a **subscriber** is subscribed to a **list**.
+
+### Subscriber-Campaign Connection
+Table name: `phplist_usermessage`
+
+Should we use a named association for this? What should it be named?
+
+
+## Messaging Context
+
+
+### Attachment
+Table name: `phplist_attachment`
+
+An attachment represents a file attached to exactly one **campaign**.
+
+### Bounce
+Table name: `phplist_boune`
+
+### BounceRegEx
+Table name: `phplist_bounceregex`
+
+### Campaign
+Table name: `phplist_message`
+
+A **campaign** is a non-personalized message on a **list** (or potentially
+multiple lists). The campaign has been created by an **administrator** owner
+and sends out email messages to multiple **subscribers**. It is stored to
+which subscribers a campaign has been sent.
+
+### CampaignBounce
+Table name: `phplist_message_bounce`
+
+### CampaignData
+Table name: `phplist_messagedata`
+
+This contains additional miscellaneous attributes for **campaigns**, such as
+Google tracking IDs, special relationships to **lists**, and alias titles.
+
+### CampaignForward
+Table name: `phplist_message_forward`
+
+This tracks details of **campaigns** which were forwarded by a recipient
+**subscriber** to someone else via an email message.
+
+### List
+Table name: `phplist_list`
+
+A **list** is something to which **subscribers** can subscribe in order to
+receive email messages from **campaigns**. Each list has exactly one **administrator**
+owner.
+
+### Template
+Table name: `phplist_template`
+
+This stores **campaign** templates. The blob contains HTML which the campaign
+content is inserted into, using special tags (AKA placeholders).
+
+### TemplateImage
+Table name: `phplist_templateimage`
+
+This contains images used in **templates**. The blob contains the image.
+
+
+## System Context
+
+### Configuration
+Table name: `phplist_config`
+
+A **configuration** item is a single value in a configuration registry.
+
+### Internationalization
+Table name: `phplist_i18n`
+
+Some background on the i18n system is here:
+https://resources.phplist.com/translations/start
+Translations are automatically imported from a repository and then accessed
+from the database and cached in the browser.
+
+### LogEntry
+Table name: `phplist_eventlog`
+
+An event **log entry** stores something that has happened on the system.
+
+What kind of events are stored?
+
+### SendProcess
+Table name: `phplist_sendprocess`
+
+This is really a "process locking" table.
+
+### UrlCache
+Table name: `phplist_urlcache`
+
+What does this do? How does this work?
+
+
+## Tracking Context
+
+### LinkTrackMl
+Table name: `phplist_linktrack_ml`
+
```

Comment:
the linktracking works as follows. All links in a campaign (m) are converted and stored in the **linktrack_forward** table. Links are not duplicated. Once there is an entry in **linktrack_forward** there reference to the campaign is stored in **linktrack_ml** (message-link). 

When a subscriber clicks this is recorded in the **linktrack_uml_click** table as "user-message-link" so that we know which subscriber clicks on which link in which campaign.